packages/expo-camera/ios/CameraViewNextModule.swift

```diff
@@ -0,0 +1,247 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+import AVFoundation
+import ExpoModulesCore
+import VisionKit
+
+struct ScannerContext {
+  var promise: Promise?
+  var delegate: Any?
+}
+
+public final class CameraViewNextModule: Module, ScannerResultHandler {
+
+  private var scannerContext: ScannerContext?
+
+  public func definition() -> ModuleDefinition {
+    Name("ExpoCameraNext")
+
+    Events("onModernBarcodeScanned")
+
+    OnCreate {
+      let permissionsManager = self.appContext?.permissions
+
+      EXPermissionsMethodsDelegate.register(
+        [
+          CameraPermissionRequester(),
+          CameraOnlyPermissionRequester(),
+          CameraMicrophonePermissionRequester()
+        ],
+        withPermissionsManager: permissionsManager
+      )
+    }
+
+    Property("modernBarcodeScannerAvailable")
+      .get { () -> Bool in
+        if #available(iOS 16.0, *) {
+          return true
+        }
+        return false
+      }
```

Comment:
If the property is just a getter, I would recommend using it like this:
```suggestion
    Property("modernBarcodeScannerAvailable") { () -> Bool in
      if #available(iOS 16.0, *) {
        return true
      }
      return false
    }
```

Reply:
One more thing: `isModernBarcodeScannerAvailable` might sound better as a property name